lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -1922,93 +1971,101 @@ void ModuleEmitter::collectNamesEmitDecls(Block &block) {
   };
 
   SmallVector<FlatBundleFieldEntry, 8> fieldTypes;
-  SmallVector<Operation *, 16> declsToEmit;
+  SmallVector<OpResult, 16> declsToEmit;
 
   for (auto &op : block) {
-    if (op.getNumResults() == 0 || isa<rtl::RTLInstanceOp>(op))
+    if (op.getNumResults() == 0)
       continue;
 
-    assert(op.getNumResults() == 1 && "firrtl only has single-op results");
-    Value result = op.getResult(0);
+    for (size_t i = 0, e = op.getNumResults(); i < e; ++i) {
```

Comment:
I'd recommend splitting this loop out to a stand-alone method that is called from within the "for op in blocks" loop.  You could land this as an NFC refactoring ahead of this patch to help with review.  This would also make the error path below cleaner (since it can just be a "return Failure", instead of messing with break statements etc).